### PR TITLE
Added DateTypePool and dataTypes in array di.xml

### PR DESCRIPTION
**Pull request by sedonik**
### Description (*)

Added Magento\CatalogGraphQl\Model\Product\Option\DateTypePool class.
Added dataTypes array (Y-m-d, Y-m-d H:i:s, Y-m-d H:i:s) in di.xml.
Implemented DI for DateTypePool with the constructor.

### Fixed Issues

1. magento/graphql-ce#761: [Customizable Options] Call to a member function format() on boolean

### Manual testing scenarios (*)

1. Create Simple Product similar to :
SKU: simple-product-1
qty: 100500
Customizable options:
type: date

2. Perform query for Customer and use header
Header:

```
{
  "Authorization": "Bearer {{ CUSTOMER_TOKEN }}"
}
```

Query:

```
mutation {
  addSimpleProductsToCart(
    input: {
      cart_id: "{{ CART_ID }}"
      cart_items: {
        data: {
          quantity: 1
          sku: "simple-product-3"
        }
        customizable_options: [
          {
            id: 1
            value_string: "12-12-12"
          }
        ]
      }
    }
  ) {
    cart {
      items {
        quantity
        product {
          sku
        }
      }
    }
  }
}
```
3. Product was added to cart

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CatalogGraphQl/etc/di.xml [hunk 1]
```diff
@@ -37,6 +37,15 @@
             </argument>
         </arguments>
     </type>
+    <type name="Magento\CatalogGraphQl\Model\Product\Option\DateTypePool">
+        <arguments>
+            <argument name="dataTypes" xsi:type="array">
+                <item name="date" xsi:type="string">Y-m-d</item>
+                <item name="date_time" xsi:type="string">Y-m-d H:i:s</item>
+                <item name="time" xsi:type="string">Y-m-d H:i:s</item>
```
Comment (fooman) [inline app/code/Magento/CatalogGraphQl/etc/di.xml:45]: Not sure if date_time and time are the same format on purpose.

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
Hi @sedonik! Thanks for your contribution, however I have few concerns regarding the expected behavior. It would be more obvious to see error message like `Invalid format provided. Please use 'Y-m-d' for 'date' type instead of formatting. The same for `date-time` and `time` https://github.com/magento/graphql-ce/issues/761#issuecomment-513943024

Could you please address this concern and update your PR? Thank you.

**Comment by lenaorobei** [comment]
@sedonik, how is your progress with this ticket, are you interested to finish it?

**Review by lenaorobei — CHANGES_REQUESTED** [review]
@sedonik, how is your progress with this ticket, are you interested to finish it?

**Comment by sedonik** [comment]
Hi @lenaorobei,

Sorry for the late response. I was on a vacation. I'm very interested to finish the task. 

I've updated PR with new [commit](https://github.com/magento/graphql-ce/pull/771/commits/7a9eedb72ffe55945f5768b7d434ec78af5a10c2).

Am I in the correct way? Pls, advice.

**Comment by lenaorobei** [comment]
@sedonik, pool is redundant here, so if date cannot be crated with `DateTime::DATETIME_PHP_FORMAT` than we need to throw a `GraphQlInputException `. We don't want to hide formatting logic and try to transform any value to the date.

**Comment by sedonik** [comment]
Sorry, it is not very clear what logic should be implemented. 
Customizable Option in our case has next types: 'date', 'date_time' and 'time'. Any of these types has its own format. That's why I created Pool of Date Types. 

Seems like it has the sense to create new [PR](https://github.com/magento/graphql-ce/pull/804).
@lenaorobei  could you take look and correct/tell if it needs to add some modifications.

Thanks.

**Comment by lenaorobei** [comment]
@sedonik yes https://github.com/magento/graphql-ce/pull/804 looks good to me.

**Pull request closed without merging**